### ISSUE TITLE
Property `__typename` does not exist on unions

### ISSUE THREAD

**Issue by jeggy** [aPureBase/KGraphQL#105]
As pointed here: https://github.com/aPureBase/KGraphQL/issues/83#issuecomment-661785451

querying for `__typename` on a union type doesn't work.

**Comment by jeggy** [comment]
Hi @Leopard2A5,

Thanks for the report, yes I think I have also seen this issue before myself.

Let's track this issue here.

**Comment by jeggy** [comment]
No, I think i know what's going wrong here.
I will take a look at it later today and report progress back here 👍

**Comment by jeggy** [comment]
@Leopard2A5 I was not able to reproduce this :confused: 

I created this test, but it does succeed: https://github.com/aPureBase/KGraphQL/blob/7fe948e20058db75ae29b1a0ab4a38d4d4658de6/kgraphql/src/test/kotlin/com/apurebase/kgraphql/integration/github/GitHubIssue105.kt

Could you take a look at that test and maybe figure out what I did different compared to what you have?

**Comment by Leopard2A5** [comment]
Hey!

It succeeds for me too! I made a slight change, embedded the ContactStatus into another type and return that type from the query, this makes the error appear:

```kotlin
data class Carrier(
    val contactStatus: ContactStatus
)

val schema = KGraphQL.schema {
    longScalar<Instant> {
        serialize = { it.toEpochMilli() }
        deserialize = { Instant.ofEpochMilli(it) }
    }
    unionType<ContactStatus>()
    query("carrier") {
        resolver { -> Carrier(ContactStatus.Onboarded(userId = "Leopard2A5")) }
    }
}

val results = schema.executeBlocking("""{
    carrier {
        contactStatus {
            ... on NotInvited {
                dummy
            }
            ... on Invited {
                invitedAt
                invitationId
            }
            ... on Onboarded {
                onboardedAt
                userId
            }
            __typename
        }
    }
}""".trimIndent()).also(::println)
```

**Comment by jeggy** [comment]
This has been fixed in version 0.15.1 👍

**Comment by Leopard2A5** [comment]
Awesome, thank you for the quick fix!
